site/docs/how-to-release.md

```diff
@@ -294,6 +294,8 @@ evolution.
 
 This release can be downloaded from: https://www.apache.org/dyn/closer.cgi/iceberg/<TARBALL NAME WITHOUT .tar.gz>/<TARBALL NAME>
 
+Release notes: https://iceberg.apache.org/releases/#XYZ-release
+
```

Comment:
I think it is good to mention the release notes link in the announce mail (like we did this time). 
Usually users used to ask about release notes when we publish the release.